matrix_addition.c

```diff
@@ -63,4 +58,14 @@ int main(int argc, char *argv[]) {
     double end_parallel = omp_get_wtime();
     double cpu_time_used_parallel = (double) (end_parallel - start_parallel);
     printf("parallel: for n=%d, p=%d, time taken=%f, speedup=%f\n", n, p, cpu_time_used_parallel, cpu_time_used_seq / cpu_time_used_parallel);
-}
\ No newline at end of file
+    
```

Comment:
Nice, it was sloppy from me to forget to free the memory I allocated for the matrices.